charts/redpanda/configmap.tpl.go

```diff
@@ -19,18 +19,767 @@ package redpanda
+	// then check it. If the flag is missing, then allow such config.
+	if !ptr.Deref(listenerEnabled, true) {
+		return false
+	}
+
+	// listener TLS enabled exist
+	if !ptr.Deref(listenerTLSEnabled, true) {
+		return false
+	}
+
+	// When TLS.Enabled is not defined, then internalTLSEnabled flag could disable TLS generation
+	if listenerTLSEnabled == nil && internalTLSEnabled != nil && !*internalTLSEnabled {
+		return false
+	}
+
+	// When TLS.Enabled is not defined, then  flag could disable TLS generation
+	if listenerTLSEnabled == nil && internalTLSEnabled == nil && !globalTLSEnabled {
+		return false
+	}
+
+	certName := internalCertName
+
+	if listenerCertName != nil {
+		certName = *listenerCertName
+	}
+
+	_, ok := certMap[certName]
+	return ok
+}
+
+func rpcListenersTLS(dot *helmette.Dot) map[string]any {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	r := values.Listeners.RPC
+
+	if r.TLS.Enabled == nil && !values.TLS.Enabled {
+		return map[string]any{}
+	}
+
+	if r.TLS.Enabled != nil && !*r.TLS.Enabled {
+		if !values.TLS.Enabled {
+			return map[string]any{}
+		}
+	}
+
+	certName := r.TLS.Cert
+	if certName == "" {
+		return map[string]any{}
+	}
+
+	return map[string]any{
+		"enabled":             true,
+		"cert_file":           fmt.Sprintf("/etc/tls/certs/%s/tls.crt", certName),
+		"key_file":            fmt.Sprintf("/etc/tls/certs/%s/tls.key", certName),
+		"require_client_auth": r.TLS.RequireClientAuth,
+		"truststore_file":     getCertificate(&values.TLS.Certs, certName),
+	}
+}
+
+func rpcListeners(dot *helmette.Dot) map[string]any {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	return map[string]any{
+		"address": "0.0.0.0",
+		"port":    values.Listeners.RPC.Port,
+	}
+}
+
+func kafkaListenersTLS(dot *helmette.Dot) []map[string]any {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	kafka := []map[string]any{}
+
+	internal := createInternalListenerTLSCfg(values.TLS.Enabled, values.Listeners.Kafka.TLS, &values.TLS.Certs)
+	if len(internal) > 0 {
+		kafka = append(kafka, internal)
+	}
+
+	for k, l := range values.Listeners.Kafka.External {
+		if !isListenerTLSCfgAvailable(
+			l.Enabled,
+			l.TLS.IsEnabled(),
+			values.Listeners.Kafka.TLS.Enabled,
+			values.TLS.Enabled,
+			values.Listeners.Kafka.TLS.Cert,
+			l.TLS.GetCert(),
+			values.TLS.Certs) {
+			continue
+		}
+
+		certName := ptr.Deref(l.TLS.GetCert(), values.Listeners.Kafka.TLS.Cert)
+
+		kafka = append(kafka, map[string]any{
+			"name":                k,
+			"enabled":             true,
+			"cert_file":           fmt.Sprintf("/etc/tls/certs/%s/tls.crt", certName),
+			"key_file":            fmt.Sprintf("/etc/tls/certs/%s/tls.key", certName),
+			"require_client_auth": ptr.Deref(l.TLS.RequireClientAuth, false),
+			"truststore_file":     getCertificate(&values.TLS.Certs, certName),
+		})
+	}
+	return kafka
+}
+
+func getCertificate(certs *TLSCertMap, certName string) string {
```

Comment:
This is an extremely misleading function name. Maybe we move this function onto the `TLS` struct and rename it to something like `getTrustStoreFilePath`?

Reply:
Thanks, changed that